File: agent-app/lib/services/auth_service.dart
import '../models/user.dart';
import 'api_service.dart';

class AuthService {
  final ApiService _apiService = ApiService();

  // Login
  Future<User> login(String email, String password) async {
    try {
      final response = await _apiService.post('/auth/login', {
        'email': email,
        'password': password,
      });

      // Save token
      final token = response['token'];
      await _apiService.saveToken(token);

      // Return user
      final userData = response['user'];
      return User.fromJson(userData);
    } catch (e) {
      throw Exception('Login failed: $e');
    }
  }

  // Logout
  Future<void> logout() async {
    try {
      await _apiService.post('/auth/logout', {});
    } catch (e) {
      // Ignore error on logout
    } finally {
      await _apiService.removeToken();
    }
  }

  // Get current authenticated user
  Future<User> getCurrentUser() async {
    try {
      final response = await _apiService.get('/auth/me');
      return User.fromJson(response['user']);
    } catch (e) {
      throw Exception('Failed to get user data: $e');
    }
  }

  // Check if user is logged in
  Future<bool> isLoggedIn() async {
    final token = await _apiService.getToken();
    return token != null;
  }
}

File: admin-panel/src/services/authService.js
import api from './api';

const authService = {
  // Login
  login: async (email, password) => {
    try {
      const response = await api.post('/login', { email, password });
      const { token, user } = response.data;

      // Store token and user info in localStorage
      localStorage.setItem('auth_token', token);
      localStorage.setItem('user', JSON.stringify(user));

      return { token, user };
    } catch (error) {
      throw error.response?.data?.message || 'Login failed';
    }
  },

  // Logout
  logout: async () => {
    try {
      await api.post('/logout');
    } catch (error) {
      console.error('Logout error:', error);
    } finally {
      // Clear local storage regardless of API response
      localStorage.removeItem('auth_token');
      localStorage.removeItem('user');
    }
  },

  // Get current authenticated user
  getCurrentUser: async () => {
    try {
      const response = await api.get('/me');
      const user = response.data.user;

      // Update user in localStorage
      localStorage.setItem('user', JSON.stringify(user));

      return user;
    } catch (error) {
      throw error.response?.data?.message || 'Failed to get user info';
    }
  },

  // Check if user is authenticated
  isAuthenticated: () => {
    return !!localStorage.getItem('auth_token');
  },

  // Get stored user from localStorage
  getStoredUser: () => {
    const user = localStorage.getItem('user');
    return user ? JSON.parse(user) : null;
  },

  // Check user role
  hasRole: (role) => {
    const user = authService.getStoredUser();
    return user?.role === role;
  },

  // Check if user is super admin
  isSuperAdmin: () => {
    const user = authService.getStoredUser();
    return user?.role === 'super_admin';
  },

  // Check if user is admin (regular admin, not super admin)
  isAdmin: () => {
    const user = authService.getStoredUser();
    return user?.role === 'admin';
  },

  // Check if user is agent
  isAgent: () => {
    const user = authService.getStoredUser();
    return user?.role === 'agent';
  },
};

export default authService;
